Tasks/Gradle/testlib/tasklib-wrapper.ts

```diff
@@ -0,0 +1,158 @@
+// Redirects calls to vsts-task-lib to testlib if in a test environment.
+import tasklib = require('vsts-task-lib/task');
+import testlib = require('./testlib-main');
+
+// TODO: should go away when task lib
+export interface EndpointAuthorization {
+    parameters:{
+        [key: string]: string;
+    };
+    scheme:string;
+}
+
+export var tl:any = tasklib;
+
+/* Switch to test from production */
+export function enableTesting():void {
+    tl = testlib;
+}
+
+export function isInTest() {
+    return tl == testlib;
+}
+
+/*  Available wrapped functions */
+
+/* Implementation:
+ (current method name)   (call with the current arguments and context)
+ return tl..apply(this, arguments);
+ */
+
+export function debug(input:string) {
+    return tl.debug.apply(this, arguments);
```

Comment:
It would have been better to use arguments.callee, but this is not allowed in strict mode.